src/client/options/mod.rs

```diff
@@ -702,6 +714,87 @@ impl ClientOptions {
         }
         Ok(())
     }
+
+    /// Merges these options with another set of ClientOptions (excluding hosts). If a field is
+    /// present on both options, favors the option present on self.
+    #[cfg(test)]
+    pub(crate) fn merge(&mut self, other: ClientOptions) {
```

Comment:
this could possibly just be replaced with a usage of the `resolve_options` macro we have defined in `src/options.rs`. The macro expects the second argument to be an `Option` though, so I'm not sure if the current mutable approach would work. You could instead return a new options struct that's a result of the merging.

Check out `src/coll/mod.rs` for some usages of it.

Reply:
I think this would work, but it would still require hard-coding the fields to merge since the third param to the macro is a list of options to check for. I could just pass in a list of all the options present on `ClientOptions`, but that's a pretty long list and is not super maintainable if more options get added in the future. What are your thoughts?

Reply:
I presume by the existence of `serde` that there's some way to iterate over the fields of a struct which is what I'd actually like to do here, but I'm not well-versed enough in macros to tell if that's feasible -- some googling made it seem like it would be tricky.

Reply:
I was imagining it would be a giant list actually. The code as written is basically a giant list itself, just with some extra stuff for checking if things are `None` or not. I agree that either one isn't really maintainable, but without a macro for enumerating all the fields, I'm not sure how else to implement this.

Reply:
To do this automatically with a macro, you'd likely have to write a procedural macro (which is more complex than macros like `resolve_options!` or `doc!`; they need to be written in a separate crate). I think that's probably more work than it's worth; since this is test code, I'm fine with hard-coding and then updating whenever we need to.

Reply:
agreed, I'll update